src/core/Akka.TestKit/TestKitBase_Receive.cs

```diff
@@ -40,8 +40,9 @@ public object FishForMessage(Predicate<object> isMessage, TimeSpan? max = null,
         /// <param name="isMessage">The is message.</param>
         /// <param name="max">The maximum.</param>
         /// <param name="hint">The hint.</param>
+        /// <param name="allMessages">If null then will be ignored. If not null then will be filled with all the messages until <paramref name="isMessage"/> returns <c>true</c></param>
         /// <returns>Returns the message that <paramref name="isMessage"/> matched</returns>
-        public T FishForMessage<T>(Predicate<T> isMessage, TimeSpan? max = null, string hint = "")
+        public T FishForMessage<T>(Predicate<T> isMessage, TimeSpan? max = null, string hint = "", List<object> allMessages = null)
```

Comment:
In order to make this change binary compatible - you need to introduce a new overload here, rather than an optional parameter.

Reply:
As can be seen in https://github.com/akkadotnet/akka.net/pull/5552/commits/4677480c592eb5ad10f1341ca60252dd7286831e, I got a build error: "Error	CS0121	The call is ambiguous between the following methods or properties: 'TestKitBase.FishForMessage<T>(Predicate<T>, TimeSpan?, string)' and 'TestKitBase.FishForMessage<T>(Predicate<T>, TimeSpan?, string, List<object>)'	Akka.TestKit.Tests (net471), Akka.TestKit.Tests (net6.0), Akka.TestKit.Tests (netcoreapp3.1)	C:\Users\brahm\source\repos\akka.net\src\core\Akka.TestKit.Tests\TestKitBaseTests\ReceiveTests.cs	54"

Reply:
```csharp
        [Fact]
        public void FishForMessage_should_return_matched_message()
        {
            TestActor.Tell(1);
            TestActor.Tell(2);
            TestActor.Tell(10);
            TestActor.Tell(20);
            FishForMessage<int>(i => i >= 10).ShouldBe(10); /// <<<<< ""Error CS0121 The call is ambiguous..."
        }


```

Reply:
Might need to rename the method then. Either way, can't break binary compat.